File: app/pages/organizations/_organization_id/index/props.ts
import { AppLayoutInfo } from "layouts/app_layout"

export type Props = {
	appLayoutInfo: AppLayoutInfo
	id: string
	members: Array<{
		email: string
		id: string
		isAdmin: boolean
	}>
	name: string
	repos: Array<{
		id: string
		title: string
	}>
	userId: string
}

File: app/pages/organizations/_organization_id/index/props.rs
use tangram_app_common::organizations::Member;
use tangram_app_layouts::app_layout::AppLayoutInfo;

#[derive(serde::Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Props {
	pub app_layout_info: AppLayoutInfo,
	pub id: String,
	pub members: Vec<Member>,
	pub name: String,
	pub repos: Vec<Repo>,
	pub user_id: String,
}

#[derive(serde::Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Repo {
	pub id: String,
	pub title: String,
}
